File: client/shared/src/commonMain/kotlin/com/chefstol/app/domain/Slot.kt
package com.chefstol.app.domain

import kotlinx.datetime.Instant

enum class SlotStatus { SCHEDULED, CANCELLED }

// data-model.md → Slot. totalSeats/freeSeats are the source of truth for capacity — the
// 8/12-seat business rule is enforced on the backend when a slot is created, never re-derived
// or hardcoded here (FR-12).
data class Slot(
    val id: String,
    val program: Program,
    val chef: Chef,
    val startAt: Instant,
    val totalSeats: Int,
    val freeSeats: Int,
    val price: Int,
    val status: SlotStatus,
    val cancellationReason: String?,
)

File: backend/internal/domain/slot.go
package domain

import "time"

// SlotStatus — slots/models.yaml#/components/schemas/SlotStatus.
type SlotStatus string

const (
	SlotScheduled SlotStatus = "scheduled"
	SlotCancelled SlotStatus = "cancelled"
)

// Slot — data-model.md → Slot. TotalSeats/FreeSeats are the source of truth for capacity;
// the 8/12-seat business rule is enforced when a slot is created, not re-derived here or on
// the client (FR-12).
type Slot struct {
	ID                 string
	Program            Program
	Chef               Chef
	StartAt            time.Time
	TotalSeats         int
	FreeSeats          int
	Price              int
	Status             SlotStatus
	CancellationReason *string
}
